### PR TITLE
Add warn test

### PR DESCRIPTION
**Pull request by Kampfkarren**

### CODE REVIEW

**Comment by Kampfkarren** [comment]
This is an issue. I can fix the whitespace, but the warning of setting readonly io.stderr (which it isn't, at least on my environment) is kinda the entire point. I wonder if I can get around that somehow.

**Comment by LPGhatguy** [comment]
That's cool, overriding `io.stderr` is a clever way to test these things.

File: lib/functions/warn_spec.lua [hunk 1]
```diff
@@ -2,7 +2,22 @@ local warn = import("./warn")
 
 describe("functions.warn", function()
 	it("should be a function", function()
-		-- Not completely sure how to test this function yet.
 		assert.is_function(warn)
 	end)
+
+	it("should warn", function()
+	   local oldErr = io.stderr
```
Comment (LPGhatguy) [inline lib/functions/warn_spec.lua]: It looks like this block has mixed indentation between tabs and spaces!

Comment (Kampfkarren) [reply lib/functions/warn_spec.lua]: Blame my environment, heh. I'll fix it.

**Pull request merged**